
处理异常主要分为两种
try catch finally  和  throws

第一种
try{

    代码/可能有异常

}catch(Exception e){

    //捕获到异常
    //1、当异常发生时
    //2、系统讲异常封装成Exception对象e,传递给catch
    //3、得到异常后，程序员自己处理
    //4、注意，如果没有发生异常 catch代码块不执行

}finally{

    //1、不管try代码块是否异常，始终要执行finally
    //2、所以通常将释放资源的代码放在finally

}


第二种
throws具体看图片
